javalib/src/main/scala/java/lang/Math.scala

```diff
@@ -131,23 +131,27 @@ object Math {
   @inline def log1p(a: scala.Double): scala.Double =
     cmath.log1p(a)
 
-  @inline def max(a: scala.Double, b: scala.Double): scala.Double =
-    `llvm.maxnum.f64`(a, b)
+  @inline def max(a: scala.Double, b: scala.Double): scala.Double = {
+    if (a.isNaN || b.isNaN) Double.NaN else `llvm.maxnum.f64`(a, b)
+  }
 
-  @inline def max(a: scala.Float, b: scala.Float): scala.Float =
-    `llvm.maxnum.f32`(a, b)
+  @inline def max(a: scala.Float, b: scala.Float): scala.Float = {
+    if (a.isNaN || b.isNaN) Float.NaN else `llvm.maxnum.f32`(a, b)
+  }
 
   @inline def max(a: scala.Int, b: scala.Int): scala.Int =
     if (a > b) a else b
 
   @inline def max(a: scala.Long, b: scala.Long): scala.Long =
     if (a > b) a else b
 
-  @inline def min(a: scala.Double, b: scala.Double): scala.Double =
-    `llvm.minnum.f64`(a, b)
+  @inline def min(a: scala.Double, b: scala.Double): scala.Double = {
+    if (a.isNaN || b.isNaN) Double.NaN else `llvm.minnum.f64`(a, b)
+  }
```

Comment:
We typically don't use curlies if body is single-line.

Reply:
Changes made as requested